npc/jobs/valkyrie.txt

```diff
@@ -217,45 +217,165 @@ yuno_in02,88,164,5	script	Metheus Sylphe#Library	2_M_SAGE_B,{
 			mes "[Metheus Sylphe]";
 			mes "Welcome to the Library of the Schweicherbil Magic Academy.";
 			mes "I assume you have come here";
-			mes "to read the 'Book of Ymir.'";
-			next;
-			mes "[Metheus Sylphe]";
-			mes "Unfortunately, the original copy of the book has been damaged over time. We currently only allow the public to view a copy of the book.";
-			next;
-			mes "[Metheus Sylphe]";
-			mes "Also, in order to preserve the original 'Book of Ymir,' we have decided to accept donations from people who wish to read the copy we have provided.";
-			next;
+			if (RENEWAL) {
+				if (questprogress(5160)) { // Quest #5160 - Library Mistake
+					mes "to confirm the status of the library guards report?";
+					next;
+					switch (select((!countitem(6586)? "Tell me the report.":""), (countitem(6586)? "I found the book.":""), "Donate Zeny")) {
```

Comment:
Donation case here should be handled on it's own and not fall through to the other one as it has different text.

Reply:
updated